sorc/ncep_post.fd/MISCLN.f

```diff
@@ -3422,11 +3457,361 @@ SUBROUTINE MISCLN
                endif
         ENDIF
       ENDIF
+
+!
+! --- Effective (inflow) Layer (EL)
+!
+      IF ( EL_SCHEME > 0 ) THEN
+        ALLOCATE(EL_BASE(IM,JSTA_2L:JEND_2U))
+        ALLOCATE(EL_TOPS(IM,JSTA_2L:JEND_2U))
+        ALLOCATE(FOUND_BASE(IM,JSTA_2L:JEND_2U))
+        ALLOCATE(FOUND_TOPS(IM,JSTA_2L:JEND_2U))
+!$omp parallel do private(i,j)
+        DO J=JSTA,JEND
+          DO I=1,IM
+            EL_BASE(I,J) = LM
+            EL_TOPS(I,J) = LM
+            FOUND_BASE(I,J) = .FALSE.
+            FOUND_TOPS(I,J) = .FALSE.
+          ENDDO
+        ENDDO
+      END IF
+!
+      IF ( EL_SCHEME == 1 ) THEN
+        ITYPE = 2
+        DPBND = 0.
+
+        DO L = LM, 1, -1
+
+!         SET AIR PARCELS FOR LEVEL L
+!$omp parallel do private(i,j)
+          DO J=JSTA,JEND
+            DO I=1,IM
+              EGRID1(I,J) = -H99999
+              EGRID2(I,J) = -H99999
+              IDUMMY(I,J) = 0
+              P1D(I,J)    = PMID(I,J,L)
+              T1D(I,J)    = T(I,J,L)
+              Q1D(I,J)    = Q(I,J,L)
+            ENDDO
+          ENDDO
+
+!---      CALCULATE CAPE/CIN FOR ALL AIR PARCELS on LEVEL L
+          IF (debugprint) WRITE(1000+ME,'(1x,A,I2.2,2x,A,I3)')          &
+           'NEW EL_SCHEME:', EL_SCHEME,                                 &
+           '   CALCULATING CAPE/CINS ON LEVEL:',L
+          CALL CALCAPE(ITYPE,DPBND,P1D,T1D,Q1D,IDUMMY,EGRID1,           &
+                          EGRID2,EGRID3,EGRID4,EGRID5)
+
+!---      CHECK CAPE/CIN OF EACH AIR PARCELS WITH EL CRITERIA
+!$omp parallel do private(i,j)
+          DO J=JSTA,JEND
+            DO I=1,IM
+              IF ( .NOT. FOUND_BASE(I,J) ) THEN
+                IF ( EGRID1(I,J) >= 100. .AND. EGRID2(I,J) >= -250. ) THEN
+                  EL_BASE(I,J) = L
+                  FOUND_BASE(I,J) = .TRUE.
+                ELSE
+                  EL_BASE(I,J) = LM
+                  FOUND_BASE(I,J) = .FALSE.
+                END IF 
+              ELSE
+                IF ( .NOT. FOUND_TOPS(I,J) ) THEN
+                  IF ( EGRID1(I,J) < 100. .OR. EGRID2(I,J) < -250. ) THEN
+                    EL_TOPS(I,J) = L + 1
+                    FOUND_TOPS(I,J) = .TRUE.
+                  ELSE
+                    EL_TOPS(I,J) = LM
+                    FOUND_TOPS(I,J) = .FALSE.
+                  END IF 
+                END IF 
+              END IF
+            ENDDO
+          ENDDO
+
+        END DO ! L
+!
+      ELSE IF ( EL_SCHEME == 2 ) THEN
```

Comment:
When I developed the code to search for Effective-Layer, I made two schemes. So the two schemes could compare to each other and verify against each other. After the testing in HWT SFE 2021, it seems that Scheme 1 had been working well and SPC was satisfied with it. 
So to make the code management and maintenance simple and easy, I suggest to set Scheme 1 as the only scheme for Effective-Layer, and remove the code related to Scheme 2 (in this MISCLN.f, and in UPP_PHYSICS.f and in TTBLEX.f). Just my suggestion. These code still can be kept.

Reply:
@EdwardColon-NOAA and @GangZhao-NOAA If scheme 2 is used only by 3DRTMA, I would agree with Gang that keeping scheme 1 only.